src/box/lua/schema.lua

```diff
@@ -2340,8 +2340,21 @@ end
 
 box.internal.check_select_opts = check_select_opts -- for net.box
 
+local check_select_args_rl = log.internal.ratelimit:new()
+local function check_select_args(index, key, opts)
+    local rl = check_select_args_rl
+
+    if index.space_id >= 512 and
+       (type(key) == 'nil' or (type(key) == 'table' and next(key) == nil)) and
+       (opts == nil or not opts.fullscan) then
+        rl:log_crit('empty or nil `select` call on user space with id=%d\n %s',
+                    index.space_id, debug.traceback())
```

Comment:
`debug.traceback()` is not a cheap call. I would not call it for messages that will be suppressed later. I don't see a problem to support a callable object as an argument and pass a function here.